google/cloud/pubsub/internal/defaults_test.cc

```diff
@@ -207,6 +207,55 @@ TEST(OptionsTest, UserSetSubscriberOptions) {
   EXPECT_EQ(6, opts.get<pubsub::MaxConcurrencyOption>());
 }
 
+TEST(OptionsTest, DefaultSubscriberOnly) {
```

Comment:
I think what you are saying is that because (1) we have a good test for whatever generates `sub_opts`, and (2) we have a good test for `internal::MakeOptions()`, and (3) we have a good test for `MergeOptions()`, *and* (4) we know how these options are composed, we can **infer** that the credentials argument will be preserved.

I am asserting that it would be better to have a direct test, instead of a series of tests and some (sound I admit) reasoning as to why that is enough.  This is true in general, direct tests are better (but more tedious) than tests of the parts and knowledge about how the parts are composed.  This is because any changes about how the parts are composed can go undetected by tests for just each part.

In any case, I am not inclined to argue too strongly for this.  Think it through and make your own call.